virt-v2v/cold/entrypoint.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+const (
+	OVA     = "ova"
+	vSphere = "vSphere"
+	DIR     = "/var/tmp/v2v"
+	FS      = "/mnt/disks/disk[0-9]*"
+	Block   = "/dev/block[0-9]*"
+	VDDK    = "/opt/vmware-vix-disklib-distrib"
+)
+
+var (
+	xmlFilePath string
+	server      *http.Server
+)
+
+func main() {
+	virtV2vArgs := []string{"virt-v2v", "-v", "-x"}
+	source := os.Getenv("V2V_source")
+
+	requiredEnvVars := map[string][]string{
+		vSphere: {"V2V_libvirtURL", "V2V_secretKey", "V2V_vmName"},
+		OVA:     {"V2V_diskPath", "V2V_vmName"},
+	}
+
+	if envVars, ok := requiredEnvVars[source]; ok {
+		if !checkEnvVariablesSet(envVars...) {
+			fmt.Printf("Following environment variables need to be defined: %v\n", envVars)
+			os.Exit(1)
+		}
+	}
+
+	switch source {
+	case vSphere:
+		virtV2vArgs = append(virtV2vArgs, "-i", "libvirt", "-ic", os.Getenv("V2V_libvirtURL"))
+		pwFilePath := fmt.Sprintf("%s/vmware.pw", DIR)
+		if err := os.WriteFile(pwFilePath, []byte(os.Getenv("V2V_secretKey")), 0600); err != nil {
+			fmt.Printf("Failed to write to disk: %v\n", err)
+			os.Exit(1)
+		}
+		virtV2vArgs = append(virtV2vArgs, "-ip", pwFilePath)
+
+		if _, err := os.Stat(VDDK); err == nil {
+			virtV2vArgs = append(virtV2vArgs,
+				"-it", "vddk",
+				"-io", fmt.Sprintf("vddk-libdir=%s", VDDK),
+				"-io", fmt.Sprintf("vddk-thumbprint=%s", os.Getenv("V2V_thumbprint")),
+			)
+		}
+		virtV2vArgs = append(virtV2vArgs, "--", os.Getenv("V2V_vmName"))
```

Comment:
I thought this should be the last argument for the command...

Reply:
I checked again, I see this arg comes at the end of the flow: https://github.com/kubev2v/forklift/blob/0e672a487e399e2ca96790e75afa0c06898ab93d/virt-v2v/cold/entrypoint#L94
am I missing somthing?

Reply:
@bkhizgiy right but later we run `virtV2vArgs = append(virtV2vArgs, "-o", "local", "-os", DIR)`

Reply:
@ahadas yeah the order here is reversed, I'm testing now the changes with live migration and uploading the final version.